server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduleTaskService.java

```diff
@@ -287,58 +272,9 @@ public Page<ScheduleTaskOverview> getScheduleTaskListResp(Pageable pageable, Lon
-
-        Specification<ScheduleTaskEntity> specification =
-                Specification.where(ScheduleTaskSpecs.jobNameIn(scheduleMap.keySet()))
-                        .and(ScheduleTaskSpecs.idEquals(params.getId()))
-                        .and(ScheduleTaskSpecs.statusIn(params.getStatuses()))
-                        .and(ScheduleTaskSpecs.fireTimeLate(params.getStartTime()))
-                        .and(ScheduleTaskSpecs.fireTimeBefore(params.getEndTime()));
-
-        Page<ScheduleTask> scheduleTaskPage = scheduleTaskRepository.findAll(specification, pageable).map(
-                scheduleTaskMapper::entityToModel);
-
-        if (scheduleTaskPage.isEmpty()) {
-            return Page.empty();
-        }
-
-        // get creator info
-        Set<Long> creatorIds = params.getSchedules().stream().map(Schedule::getCreatorId).collect(Collectors.toSet());
-        Map<Long, List<UserEntity>> users = userRepository.findByIdIn(creatorIds).stream().collect(
-                Collectors.groupingBy(UserEntity::getId));
-
-        // get database info
-        Set<Long> databaseIds = params.getSchedules().stream().map(Schedule::getDatabaseId).collect(Collectors.toSet());
-        Map<Long, Database> databaseMap = scheduleResponseMapperFactory.getDatabaseInfoByIds(databaseIds).stream()
-                .collect(Collectors.toMap(Database::getId, Function.identity()));
-
-        // get job result json
-        List<Long> jobIds = scheduleTaskPage.getContent().stream().map(ScheduleTask::getJobId).filter(Objects::nonNull)
-                .collect(Collectors.toList());
-
-        Map<Long, String> resultMap = jobRepository.findAllById(jobIds).stream()
-                .filter(jobEntity -> jobEntity.getResultJson() != null)
-                .collect(Collectors.toMap(JobEntity::getId, JobEntity::getResultJson));
-
-        return scheduleTaskPage.map(task -> {
-            Schedule schedule = scheduleMap.get(task.getJobName());
-            ScheduleTaskListOverview overview = ScheduleTaskListOverviewMapper.map(task);
-            overview.setScheduleName(schedule.getName());
-            overview.setCreator(new InnerUser(users.get(schedule.getCreatorId()).get(0), null));
-            if (schedule.getType() == ScheduleType.SQL_PLAN) {
-                SqlPlanAttributes attribute = new SqlPlanAttributes();
-                attribute.setDatabaseInfo(databaseMap.get(schedule.getDatabaseId()));
-                attribute.setTaskResult(JsonUtils.fromJson(resultMap.get(task.getJobId()), SqlPlanTaskResult.class));
-                Map<Long, String> id2Attributes = new HashMap<>();
-                id2Attributes.put(task.getId(), JsonUtils.toJson(attribute));
-                overview.setAttributes(JSON.parseObject(id2Attributes.get(task.getId())));
-            }
-            return overview;
-        });
+        return scheduleTaskRepository.find(pageable, params);
```

Comment:
why should we need this method rather than using repo directly?

Reply:
fixed